x/stakeibc/keeper/icacallbacks_detokenize_test.go

```diff
@@ -28,22 +54,32 @@ func (s *KeeperTestSuite) SetupTestDetokenizeCallback() []byte {
 	})
 	s.Require().NoError(err, "no error expected when marshalling callback args")
 
-	return callbackBz
+	return DetokenizeCallbackTestCase{
+		callbackBz:                  callbackBz,
+		expectedValidatorDelegation: expectedValidatorDelegation,
+		expectedTotalDelegation:     expectedTotalDelegation,
+	}
 }
 
 func (s *KeeperTestSuite) TestDetokenizeCallback_Successful() {
-	callbackBz := s.SetupTestDetokenizeCallback()
+	tc := s.SetupTestDetokenizeCallback()
 
 	// Call the callback with a successful response
 	ackSuccess := &icacallbackstypes.AcknowledgementResponse{
 		Status: icacallbackstypes.AckResponseStatus_SUCCESS,
 	}
-	err := keeper.DetokenizeCallback(s.App.StakeibcKeeper, s.Ctx, channeltypes.Packet{}, ackSuccess, callbackBz)
+	err := keeper.DetokenizeCallback(s.App.StakeibcKeeper, s.Ctx, channeltypes.Packet{}, ackSuccess, tc.callbackBz)
 	s.Require().NoError(err, "no error expected during callback")
 
 	// Check that the deposit was removed
 	_, found := s.App.RecordsKeeper.GetLSMTokenDeposit(s.Ctx, HostChainId, LSMTokenBaseDenom)
 	s.Require().False(found, "deposit should have been removed")
+
+	// Check that the delegation was updated
+	hostZone, found := s.App.StakeibcKeeper.GetHostZone(s.Ctx, HostChainId)
+	s.Require().True(found, "host zone should have been found")
+	s.Require().Equal(tc.expectedTotalDelegation, hostZone.TotalDelegations.Int64(), "host zone total delegation")
+	s.Require().Equal(tc.expectedValidatorDelegation, hostZone.Validators[0].Delegation.Int64(), "validator delegation")
```

Comment:
Nit: could we add another validator to this test setup and here check that its value remains unchanged?

Reply:
done

Reply:
sweet, resolving!